sql/glam-fenix-dev/glam_etl/firefox_desktop_glam_nightly__clients_scalar_aggregates_v1/query.sql

```diff
@@ -0,0 +1,258 @@
+  SELECT
+    submission_date,
+    client_id,
+    ping_type,
+    os,
+    app_version,
+    app_build_id,
+    channel,
+    metric,
+    metric_type,
+    key,
+    agg_type,
+    value
+  FROM
+    filtered_date_channel
+  CROSS JOIN
+    UNNEST(scalar_aggregates)
+  WHERE
+    value IS NOT NULL
+),
+version_filtered_new AS (
+  SELECT
+    submission_date,
+    scalar_aggs.client_id,
+    scalar_aggs.ping_type,
+    scalar_aggs.os,
+    scalar_aggs.app_version,
+    scalar_aggs.app_build_id,
+    scalar_aggs.channel,
+    metric,
+    metric_type,
+    key,
+    agg_type,
+    value
+  FROM
+    filtered_aggregates AS scalar_aggs
+  LEFT JOIN
+    glam_etl.firefox_desktop_glam_nightly__latest_versions_v1
+  USING
+    (channel)
+  WHERE
+      -- allow for builds to be slighly ahead of the current submission date, to
+      -- account for a reasonable amount of clock skew
+    mozfun.glam.build_hour_to_datetime(app_build_id) < DATE_ADD(@submission_date, INTERVAL 3 day)
+      -- only keep builds from the last year
+    AND mozfun.glam.build_hour_to_datetime(app_build_id) > DATE_SUB(
+      @submission_date,
+      INTERVAL 365 day
+    )
+    AND app_version > (latest_version - 3)
+),
+scalar_aggregates_new AS (
+  SELECT
+    client_id,
+    ping_type,
+    os,
+    app_version,
+    app_build_id,
+    channel,
+    metric,
+    metric_type,
+    key,
+    agg_type,
+    --format:off
+    CASE agg_type
+      WHEN 'max' THEN max(value)
+      WHEN 'min' THEN min(value)
+      WHEN 'count' THEN sum(value)
+      WHEN 'sum' THEN sum(value)
+      WHEN 'false' THEN sum(value)
+      WHEN 'true' THEN sum(value)
+    END AS value
+    --format:on
+  FROM
+    version_filtered_new
+  WHERE
+    -- avoid overflows from very large numbers that are typically anomalies
+    value <= POW(2, 40)
+  GROUP BY
+    client_id,
+    ping_type,
+    os,
+    app_version,
+    app_build_id,
+    channel,
+    metric,
+    metric_type,
+    key,
+    agg_type
+),
+filtered_new AS (
+  SELECT
+    client_id,
+    ping_type,
+    os,
+    app_version,
+    app_build_id,
+    channel,
+    ARRAY_AGG((metric, metric_type, key, agg_type, value)) AS scalar_aggregates
+  FROM
+    scalar_aggregates_new
+  GROUP BY
+    client_id,
+    ping_type,
+    os,
+    app_version,
+    app_build_id,
+    channel
+),
+filtered_old AS (
+  SELECT
+    scalar_aggs.client_id,
+    scalar_aggs.ping_type,
+    scalar_aggs.os,
+    scalar_aggs.app_version,
+    scalar_aggs.app_build_id,
+    scalar_aggs.channel,
+    scalar_aggregates
+  FROM
+    glam_etl.firefox_desktop_glam_nightly__clients_scalar_aggregates_v1 AS scalar_aggs
+  LEFT JOIN
+    glam_etl.firefox_desktop_glam_nightly__latest_versions_v1
+  USING
+    (channel)
+  WHERE
+      -- allow for builds to be slighly ahead of the current submission date, to
+      -- account for a reasonable amount of clock skew
+    mozfun.glam.build_hour_to_datetime(app_build_id) < DATE_ADD(@submission_date, INTERVAL 3 day)
```

Comment:
Does this work? I think you said the `app_build_id` is in seconds for FOG?

Reply:
Yes by the time it reaches to this function the app_build_id is already truncated to hour (yyyymmddhh)